dev/tools/java_and_objc_doc.dart

```diff
@@ -63,10 +64,30 @@ Future<void> generateDocs(String url, String docName, String checkFile) async {
     }
   }
 
+  /// If object then copy files to old location if the archive is using the new location.
```

Comment:
can you give me some context on why we have a new and old location? are the contents different?

Reply:
The contents are the same but the folder structure added an objectc_docs folder and puts all the contents there.

The context of that change is that with recipes v2 migration we are moving the generation of the zip files from the recipes to the engine and having the additional folder helps us prevent adding additional files to the zip if for some reason a nonempty directory is passed to the tools/gen_objcdoc.sh script.

After removing the migration to engine v2 is complete we can get rid of this additional complexity.

Reply:
SGTM